src/main/java/filter/expression/Qualifier.java

```diff
@@ -136,28 +133,74 @@ public static boolean process(IModel model, FilterExpression expr, TurboIssue is
         return exprWithNormalQualifiers.isSatisfiedBy(model, issue, new MetaQualifierInfo(metaQualifiers));
     }
 
-    private static Optional<Integer> getCurrentMilestoneIndex(List<TurboMilestone> allMilestones) {
-        if (allMilestones.isEmpty()) {
-            return Optional.empty();
+    /**
+     * Get all milestones that can be aliased with current-[n] to current+[n]
+     */
+    private static List<TurboMilestone> getMilestonesWithinAliasRange(List<TurboMilestone> milestonesOfRepos) {
+        List<TurboMilestone> milestones = new ArrayList<>();
+        // add milestones with due date first
+        milestones.addAll(milestonesOfRepos.stream()
+                .filter(ms -> ms.getDueDate().isPresent())
+                .collect(Collectors.toList()));
+
+        // Special case: if there is only one open milestone, it should be included regardless of whether
+        // it has due date or not.
+        // In this case, if it is not included already (the open milestone does not
+        // have due date), add it to the milestone list.
+        List<TurboMilestone> openMilestones = TurboMilestone.getOpenMilestones(milestonesOfRepos);
+        if (openMilestones.size() == 1) {
+            TurboMilestone openMilestone = openMilestones.get(0);
+            boolean hasDueDate = openMilestone.getDueDate().isPresent();
+            if (!hasDueDate) {
+                milestones.add(openMilestone);
+            }
         }
 
-        int currentIndex = 0;
+        return milestones;
+    }
 
-        for (TurboMilestone checker : allMilestones) {
-            boolean overdue = checker.getDueDate().isPresent() &&
-                    checker.getDueDate().get().isBefore(LocalDate.now());
-            boolean relevant = !(overdue && checker.getOpenIssues() == 0);
+    /**
+     * Returns a map where the key denotes offset from "current" milestone.
+     * i.e. 0 for "current" milestone, -1 for "current-1" milestone, etc.
+     */
+    private static Map<Integer, TurboMilestone> getMilestoneAliasIndex(List<TurboMilestone> milestones) {
+        List<TurboMilestone> sortedMilestones = TurboMilestone.getSortedMilestonesByDueDate(milestones);
+        Optional<Integer> currentMilestoneIndex = getCurrentMilestoneIndex(sortedMilestones);
 
-            if (checker.isOpen() && relevant) {
-                return Optional.of(currentIndex);
-            }
+        return IntStream
+                .range(0, sortedMilestones.size())
+                .boxed()
+                .collect(Collectors.toMap(
+                        i -> i - currentMilestoneIndex.get(),
```

Comment:
Will this be safe if `currentMilestoneIndex` is empty? Maybe cover it with a test as well.